R/filter.R

```diff
@@ -33,3 +34,22 @@ porcelain_filter_metadata <- function(req, res) {
   req$received_time <- now_utc()
   plumber::forward()
 }
+
+
+porcelain_filter_request_id <- function(logger) {
+  force(logger)
+  function(req, res) {
+    if ("x-request-id" %in% names(req$HEADERS)) {
```

Comment:
Chased down where the `HEADERS` actually come from: https://github.com/rstudio/httpuv/blob/ac79a51f8d354d38dc71e2b55b676e1a0647d029/src/webapplication.cpp#L169-L179

It seems to me that the keys of `HEADERS` are not normalized, therefore we wouldn't match on different case variations. Instead I think you should use `req$HTTP_X_REQUEST_ID` which will do the right thing.

See also https://github.com/rstudio/httpuv/blob/ac79a51f8d354d38dc71e2b55b676e1a0647d029/R/httpuv.R#L485-L491 and https://github.com/jeffreyhorner/Rook/blob/a5e45f751/README.md#the-environment for documentation of what's in `req`.

Reply:
So I checked this and `HEADERS` are case-insensitive. They are converted to lower case at some point so this does work with `X-Request-Id` and others. That said, I think it's nicer working with the `HTTP_X_REQUEST_ID` instead of full list of `HEADERS`. I've updated that